dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java

```diff
@@ -0,0 +1,752 @@
+    private final FolderAPI      folderAPI;
+    private final HostWebAPI     hostWebAPI;
+    private final PermissionAPI  permissionAPI;
+    private final RoleAPI        roleAPI;
+    private final HTMLPageAssetAPI htmlPageAssetAPI;
+    private final TemplateHelper templateHelper;
+
+
+    public TemplateResource() {
+
+        this(new WebResource(),
+                new PaginationUtil(new TemplatePaginator(APILocator.getTemplateAPI(),
+                        new TemplateHelper(APILocator.getPermissionAPI(), APILocator.getRoleAPI()))),
+                APILocator.getTemplateAPI(),
+                APILocator.getVersionableAPI(),
+                APILocator.getFolderAPI(),
+                APILocator.getPermissionAPI(),
+                WebAPILocator.getHostWebAPI(),
+                APILocator.getRoleAPI(),
+                APILocator.getHTMLPageAssetAPI());
+    }
+
+    @VisibleForTesting
+    public TemplateResource(final WebResource     webResource,
+                             final PaginationUtil templatePaginator,
+                             final TemplateAPI    templateAPI,
+                             final VersionableAPI versionableAPI,
+                             final FolderAPI      folderAPI,
+                             final PermissionAPI  permissionAPI,
+                             final HostWebAPI     hostWebAPI,
+                             final RoleAPI        roleAPI,
+                             final HTMLPageAssetAPI htmlPageAssetAPI) {
+
+        this.webResource    = webResource;
+        this.templateAPI    = templateAPI;
+        this.versionableAPI = versionableAPI;
+        this.folderAPI      = folderAPI;
+        this.permissionAPI  = permissionAPI;
+        this.hostWebAPI     = hostWebAPI;
+        this.roleAPI        = roleAPI;
+        this.htmlPageAssetAPI = htmlPageAssetAPI;
+        this.templateHelper = new TemplateHelper(permissionAPI, roleAPI);
+        this.paginationUtil = templatePaginator;
+    }
+
+    /**
+     * Return a list of {@link com.dotmarketing.portlets.templates.model.Template}, entity
+     * response syntax:.
+     *
+     *
+     * Url sintax:
+     * api/v1/templates?filter=filter-string&page=page-number&per_page=per-page&ordeby=order-field-name&direction=order-direction&host=host-id
+     *
+     * where:
+     *
+     * <ul>
+     * <li>filter-string: just return Template who content this pattern into its title</li>
+     * <li>page: page to return</li>
+     * <li>per_page: limit of items to return</li>
+     * <li>ordeby: field to order by</li>
+     * <li>direction: asc for upward order and desc for downward order</li>
+     * <li>host: filter by host's id</li>
+     * </ul>
+     *
+     * Url example: v1/templates?filter=test&page=2&orderby=title
+     *
+     * @param httpRequest
+     * @return
+     */
+    @GET
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
```

Comment:
Codacy found an issue: [The String literal "application/javascript" appears 4 times in this file; the first occurrence is on line 122](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)